src/harness/harnessLanguageService.ts

```diff
@@ -148,26 +148,26 @@ namespace Harness.LanguageService {
         }
 
         public getScriptInfo(fileName: string): ScriptInfo {
-            const fileEntry = this.virtualFileSystem.traversePath(fileName);
-            return fileEntry && fileEntry.isFile() ? (<Utils.VirtualFile>fileEntry).content : undefined;
+            const fileEntry = this.vfs.getFile(fileName);
+            return fileEntry ? fileEntry.metadata.get("scriptInfo") : undefined;
         }
 
         public addScript(fileName: string, content: string, isRootFile: boolean): void {
-            this.virtualFileSystem.addFile(fileName, new ScriptInfo(fileName, content, isRootFile));
+            this.vfs.addFile(fileName, content, { overwrite: true }).metadata.set("scriptInfo", new ScriptInfo(fileName, content, isRootFile));
         }
 
         public editScript(fileName: string, start: number, end: number, newText: string) {
-            const script = this.getScriptInfo(fileName);
-            if (script !== undefined) {
+            const file = this.vfs.getFile(fileName);
+            const script = file && file.metadata.get("scriptInfo") as ScriptInfo;

```

Comment:
The `vfs`'s construction should probably be parameterized over its intended metadata, so this access and the `set` above have strong types.

Reply:
Metadata is arbitrary and inherited. Trying to add a complex type to define the structure of metadata is an unnecessary over-complication.

Reply:
Is asking for `{scriptInfo: ScriptInfo}` as a type argument really more of a burden than casting at every `get`?